Ques.1.
A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions

# Data description
α == 0.05 (95% Confidence)

Y == Continious
X == Discrete

Is Y1 and Y2 normal ?

H0 = Y1 and Y2 are normal

H1 = Y1 and Y2 are not normal

In [3]:
#import the libraries
import pandas as pd
import scipy 
import numpy as np
from scipy import stats

In [4]:
# Load Data Set
cutlet=pd.read_csv("/content/Cutlets.csv") 

In [6]:
cutlet.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [9]:
cutlet.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [11]:
cutlet.shape

(35, 2)

In [20]:
stats.shapiro(cutlet["Unit A"])

ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)

(0.9649458527565002, 0.3199819028377533)

P value for Unit A == 0.32 > α

In [26]:
stats.shapiro(cutlet["Unit B"])

ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)

(0.9727300405502319, 0.5224985480308533).

P value for Unit B == 0.52 > α

# Model
H0 == Mean for Y1 and Y2 are equal.

H1 == Mean for Y1 and Y2 are not equal.

In [25]:
stats.ttest_rel(cutlet["Unit A"], cutlet["Unit B"])

Ttest_relResult(statistic=0.7536787225614314, pvalue=0.4562300768038412)

Ttest_relResult(statistic=0.7536787225614323,
 pvalue=0.45623007680384076)

P value of the Paired T Test is == 0.45 > α

The H0 is accepted.

Mean of both Y1 and Y2 are equal

Que.2.

A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.

Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level

 Data description
α == 0.05 (95% Confidence)

Y == Continious
X == Discrete

Is Y1, Y2, Y3 and Y4 normal ?

H0 = Y1, Y2, Y3 and Y4 are normal
H1 = Y1, Y2, Y3 and Y4 are not normal

In [31]:
# Load Data Set
LabTAT=pd.read_csv("/content/LabTAT.csv") 

In [33]:
LabTAT.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [35]:
LabTAT.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [37]:
LabTAT.shape

(120, 4)

In [39]:
stats.shapiro(LabTAT["Laboratory 1"])

ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)

(0.9901824593544006, 0.5506953597068787)

P value for Laboratory 1 == 0.55 > α

In [41]:
stats.shapiro(LabTAT["Laboratory 2"])

ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)

(0.9936322569847107, 0.8637524843215942)

P value for Laboratory 2 == 0.86 > α

In [43]:
stats.shapiro(LabTAT["Laboratory 3"])

ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)

(0.9886345267295837, 0.4205053448677063)

P value for Laboratory 3 == 0.42 > α

In [45]:
stats.shapiro(LabTAT["Laboratory 4"])

ShapiroResult(statistic=0.9913753271102905, pvalue=0.6618951559066772)

(0.9913753271102905, 0.6618951559066772)

P value for Y1  == 0.66 > α

HO is accepted. Thats is Y1, Y2, Y3, and Y4 are normal.

# Varience

H0 == Variance of all 4 laboratories are the same.

H1 == Variance of all 4 laboratories are the not same.

In [47]:
stats.levene(LabTAT["Laboratory 1"], LabTAT["Laboratory 2"], LabTAT["Laboratory 3"], LabTAT["Laboratory 4"])

LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)

LeveneResult(statistic=2.599642500418024, 
pvalue=0.05161343808309816)
P Value of Variance test is == 0.051 > α

H0 is accepted

we will perform one way Anova Test

#Model
H0 == Mean TAT for 4 laboratories equal.

H1 == Mean TAT for 4 laboratories not equal.

In [51]:
stats.stats.f_oneway(LabTAT["Laboratory 1"], LabTAT["Laboratory 2"], LabTAT["Laboratory 3"], LabTAT["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

F_onewayResult(statistic=118.70421654401437,
 pvalue=2.1156708949992414e-57)
 
P value of the One way Anova test is == 2e-16 < α

H1 is accepted.

Mean TAT for 4 laboratories not equal.

**#Ques.3.**

Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions

# Data description
α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test

In [55]:
# Load Data Set
BuyerRatio=pd.read_csv("/content/BuyerRatio.csv") 

In [56]:
BuyerRatio.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [57]:
BuyerRatio.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [58]:
BuyerRatio.shape

(2, 5)

# Chi-Square Test

In [65]:
stats.chi2_contingency([BuyerRatio["East"], BuyerRatio["West"], BuyerRatio["North"], BuyerRatio["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

P value of Chi-Square test == 0.66 > α

Thus HO is accepted.

**Product sales ratio are same for males and females**

 **Ques.4.**

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Description

α == 0.05 (95% Confidence)

Y == Discrete
X == Discrete

Since there are more than 2 variable we will perform Chi-Square test.



In [67]:
# Load Data Set
Costomer_OrderForm=pd.read_csv("/content/Costomer+OrderForm.csv") 

In [68]:
Costomer_OrderForm.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [69]:
Costomer_OrderForm.shape

(300, 4)

Chi-Square Test

In [75]:
stats.chi2_contingency([Costomer_OrderForm['Phillippines'].value_counts(), Costomer_OrderForm['Indonesia'].value_counts(), Costomer_OrderForm['Malta'].value_counts(), Costomer_OrderForm['India'].value_counts()])

(3.8589606858203545, 0.2771020991233144, 3, array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

P value of Chi-Square test == 0.277 > α

Thus HO is accepted.